backend/packages/Upgrade/src/api/services/SegmentService.ts

```diff
@@ -149,7 +164,134 @@ export class SegmentService {
     return await this.segmentRepository.deleteSegment(id, logger);
   }
 
-  public async importSegments(segments: SegmentInputValidator[], logger: UpgradeLogger): Promise<Segment[]> {
+  public async importSegments(segments: SegmentFile[], logger: UpgradeLogger): Promise<SegmentReturnObj> {
+    const importFileErrors: SegmentImportError[] = [];
+    if (this.isJSONString(segments[0].fileContent)) {
+      const parsedJsonData: SegmentInputValidator[] = [];
+      segments.forEach((segment) => {
+        const fileName = segment.fileName;
```

Comment:
This part of the code is written with an assumption that the values you are expecting will be inside the JSON. What if these values are not present? like `fileName`, `fileContent` etc.

Reply:
We are converting data we sent from frontend to this format before sending.